nest-forms-backend/src/utils/subservices/sharepoint.subservice.ts

```diff
@@ -0,0 +1,357 @@
+    const valuesForFields = getValuesForFields(
+      sharepointData,
+      { ...form, title: formTitle },
+      jsonDataExtraDataOmitted,
+      fields,
+    )
+
+    if (sharepointData.oneToOne) {
+      const oneToOneOriginalTableFields = await this.mapColumnsToFields(
+        Object.values(sharepointData.oneToOne).map(
+          (record) => record.originalTableId,
+        ),
+        accessToken,
+        sharepointData.databaseName,
+      )
+
+      await Promise.all(
+        Object.values(sharepointData.oneToOne).map(async (value) => {
+          const oneToOneFields = await this.mapColumnsToFields(
+            Object.keys(value.columnMap),
+            accessToken,
+            value.databaseName,
+          )
+          const valuesForFieldsOneToOne = getValuesForFields(
+            value,
+            { ...form, title: formTitle },
+            jsonDataExtraDataOmitted,
+            oneToOneFields,
+          )
+
+          const addedId = await this.postDataToSharepoint(
+            value.databaseName,
+            accessToken,
+            valuesForFieldsOneToOne,
+          )
+
+          valuesForFields[
+            `${oneToOneOriginalTableFields[value.originalTableId]}Id`
+          ] = addedId
+        }),
+      )
+    }
+
+    const baseId = await this.postDataToSharepoint(
+      sharepointData.databaseName,
+      accessToken,
+      valuesForFields,
+    )
+
+    if (sharepointData.oneToMany) {
+      await Promise.all(
+        Object.entries(sharepointData.oneToMany).map(async ([key, value]) => {
+          const recordsArray = getArrayForOneToMany(
+            { ...form, jsonDataExtraDataOmitted },
+            key,
+          )
+
+          await Promise.all(
+            recordsArray.map(async (record) => {
+              const foreignFields: Record<string, any> = {}
+              foreignFields[value.originalTableId] = baseId
+
+              const oneToManyFields = await this.mapColumnsToFields(
+                [
+                  ...Object.keys(value.columnMap),
+                  ...Object.keys(foreignFields),
+                ],
+                accessToken,
+                value.databaseName,
+              )
+
+              const valuesForFieldsOneToMany = getValuesForFields(
+                value,
+                { ...form, title: formTitle },
+                record,
+                oneToManyFields,
+                foreignFields,
+              )
+
+              await this.postDataToSharepoint(
+                value.databaseName,
+                accessToken,
+                valuesForFieldsOneToMany,
+              )
+            }),
+          )
+        }),
+      )
+    }
+
+    await this.prismaService.forms.update({
+      where: {
+        id: formId,
+      },
+      data: {
+        error: FormError.NONE,
+        state: FormState.PROCESSING,
+      },
+    })
+  }
+
+  private async mapColumnsToFields(
+    columns: Array<string>,
+    accessToken: string,
+    dtbName: string,
```

Comment:
`db` seems to be more common abbreviation of `database`, or is it specific for this to be `dtb`?